### PR TITLE
GPS icon now reflect fix state

### PR DESCRIPTION
**Pull request by HThuren**
GPS icon change when fixed. Start with Red = GPS configured, but no fix, change to Yellow when fix are reached.

GPS not fixed
image

GPS fixed
image

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

File: src/js/serial_backend.js
Comment (haslinghuis) [inline src/js/serial_backend.js:732]: :thinking: Maybe add this to sensor_helpers.js is a better place?

**Comment by HThuren** [comment]
@haslinghuis, nice suggestion, code moved to sensor_helpers.js
PR ready from my point of view

File: src/js/serial_backend.js
Comment (haslinghuis) [inline src/js/serial_backend.js:351]: Check the value of FC.GPS_DATA.fix here, as the value has not been queried upon opening (or is 0 as in fc.js)

Comment (HThuren) [reply src/js/serial_backend.js:351]: right, the value dosn't matter here if undefined or 0, but I change to 0

File: src/js/sensor_helpers.js
Comment (haslinghuis) [inline src/js/sensor_helpers.js:22]: This way we don't need to provide 0 values :)
```suggestion
export function sensor_status(sensors_detected = 0, gps_fix_state = 0) {
```

File: src/js/serial_backend.js
Comment (haslinghuis) [inline src/js/serial_backend.js:670]: Please remove

File: src/js/sensor_helpers.js
Comment (haslinghuis) [inline src/js/sensor_helpers.js:79]:
```suggestion
        if (gps_fix_state) {
```
should work as boolean

File: src/js/sensor_helpers.js
Comment (haslinghuis) [inline src/js/sensor_helpers.js:32]: This can potentially be optimized only updating the sensor which got updated. It's okay.
You could also look into toggleClass(class, condition), up to you.

Comment (HThuren) [reply src/js/sensor_helpers.js:32]: toggleClass ?

Comment (haslinghuis) [reply src/js/sensor_helpers.js:32]: My idea was to have one code block for `if (sensor_status.previous_sensors_detected == sensors_detected) {`
and one for `if (sensor_status.previous_gps_fix_state == gps_fix_state) {`

JQuery toggleClass(class, condition) could be a replacement for addClass(class) and removeClass(class)

**Review by haslinghuis — APPROVED** [review]

**Review by nerdCopter — APPROVED** [review]
* looks good. i cannot test, however.

File: src/js/sensor_helpers.js
Comment (HThuren) [reply src/js/sensor_helpers.js:32]: I think it will give a different logic, since sensors - including gps  - can toggle a activation; and change in gps_fix.

ToggleClass demands a boolean, then the will be a if (fix) then flag = true else flag = false; and then toggleClass('', flag)

**Review by nerdCopter — APPROVED** [review]

**Comment by nerdCopter** [comment]
i saw this, so i removed my approval
image

but then i thought it was already done, so re-applied my approval.  then it was merged.  i think @HThuren  wanted to change something before merging.

**Comment by haslinghuis** [comment]
All good. Conversation was resolved by @HThuren - we can always make a patch if needed.

**Comment by HThuren** [comment]
I was done, nice to have the merge :-)

**Comment by bact2828** [comment]
> GPS icon change when fixed. Start with Red = GPS configured, but no fix, change to Yellow when fix are reached.
> 
> GPS not fixed image
> 
> GPS fixed image

Does the red color indicate that the satellites are being locked?

Did you understand well :)

**Comment by HThuren** [comment]
hi @bact2828, the state in use are from FC, ie. FC.GPS_DATA.fix. Same status as in GPS tab:
image

**Pull request merged**
